src/libvfcfuncinstrument/libVFCFuncInstrument.cpp

```diff
@@ -286,20 +318,32 @@ struct VfclibFunc : public ModulePass {
         Type::getInt8Ty(M.getContext()),    Type::getInt32Ty(M.getContext())};
 
     // void vfc_enter_function (char*, char, char, char, char, int, ...)
-    Constant *func = M.getOrInsertFunction(
+    _LLVMFunctionType func = M.getOrInsertFunction(
         "vfc_enter_function",
         FunctionType::get(Type::getVoidTy(M.getContext()), ArgTypes, true));
 
-    func_enter = cast<Function>(func);
+#if LLVM_VERSION_MAJOR >= 9
```

Comment:
Maybe we could create a macro for this pattern ?

Reply:
Same answer as before